src/files-card-body.jsx

```diff
@@ -422,9 +422,13 @@ const Item = React.memo(function Item({ file, isSelected, isGrid }) {
                       ? "xl"
                       : "lg"} isInline
                 >
-                    {file.type === "dir" || file.to === "dir"
-                        ? <FolderIcon />
-                        : <FileIcon />}
+                    {((file.type === "dir" || file.to === "dir") && <FolderIcon />) ||
+                        (file.category &&
+                        <img
+                          width='54px' height='54px'
+                          src={'icons/' + file.category.icon + '.svg'}
+                          aria-label={file.category.name}
+                        />) || <FileIcon />}
```

Comment:
This conflicts with the work I put in yesterday where the whole ListingTable is used for both views and the svg's 
are embedded in scss.

We'll need to investigate what / how to tackle that. Either is fine but we must make sure the svg's are cached by the browser just as JS/CSS.

https://github.com/cockpit-project/cockpit-files/pull/481

Reply:
I absolutely hate this part of the patch and definitely want you or @garrett to come up with something better here.